ibm/resource_ibm_is_image.go

```diff
@@ -332,6 +361,103 @@ func imgCreate(d *schema.ResourceData, meta interface{}, href, name, operatingSy
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+	imagePrototype := &vpcv1.ImagePrototypeImageBySourceVolume{
+		Name: &name,
+	}
+	var insId string
+	imagePrototype.SourceVolume = &vpcv1.VolumeIdentity{
+		ID: &volume,
+	}
+	options := &vpcv1.GetVolumeOptions{
+		ID: &volume,
+	}
+	vol, response, err := sess.GetVolume(options)
+	if err != nil {
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error Getting Volume (%s): %s\n%s", volume, err, response)
+	}
+	volAtt := &vol.VolumeAttachments[0]
+	insId = *volAtt.Instance.ID
+	getinsOptions := &vpcv1.GetInstanceOptions{
+		ID: &insId,
+	}
+	instance, response, err := sess.GetInstance(getinsOptions)
+	if err != nil {
+		if response != nil && response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error Getting Instance (%s): %s\n%s", insId, err, response)
+	}
+	if instance != nil && *instance.Status == "running" {
+		actiontype := "stop"
+		createinsactoptions := &vpcv1.CreateInstanceActionOptions{
+			InstanceID: &insId,
+			Type:       &actiontype,
+		}
+		_, response, err = sess.CreateInstanceAction(createinsactoptions)
+		if err != nil {
+			if response != nil && response.StatusCode == 404 {
+				return nil
+			}
+			return fmt.Errorf("Error Creating Instance Action: %s\n%s", err, response)
+		}
+		_, err = isWaitForInstanceActionStop(sess, d.Timeout(schema.TimeoutDelete), insId, d)
+		if err != nil {
+			return err
+		}
+	} else if *instance.Status != "stopped" {
+		_, err = isWaitForInstanceActionStop(sess, d.Timeout(schema.TimeoutDelete), insId, d)
+		if err != nil {
+			return err
+		}
+	}
+
+	if encryptionKey, ok := d.GetOk(isImageEncryptionKey); ok {
+		encryptionKeyStr := encryptionKey.(string)
+		// Construct an instance of the EncryptionKeyReference model
+		encryptionKeyReferenceModel := new(vpcv1.EncryptionKeyIdentity)
+		encryptionKeyReferenceModel.CRN = &encryptionKeyStr
+		imagePrototype.EncryptionKey = encryptionKeyReferenceModel
+	}
+	if rgrp, ok := d.GetOk(isImageResourceGroup); ok {
+		rg := rgrp.(string)
+		imagePrototype.ResourceGroup = &vpcv1.ResourceGroupIdentity{
+			ID: &rg,
+		}
+	}
+	imagOptions := &vpcv1.CreateImageOptions{
+		ImagePrototype: imagePrototype,
+	}
+	image, response, err := sess.CreateImage(imagOptions)
+	if err != nil {
+		return fmt.Errorf("[DEBUG] Image creation err %s\n%s", err, response)
+	}
+	d.SetId(*image.ID)
+	log.Printf("[INFO] Image ID : %s", *image.ID)
+	_, err = isWaitForImageAvailableForVolume(sess, d.Id(), d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return err
+	}
+	v := os.Getenv("IC_ENV_TAGS")
+	if _, ok := d.GetOk(isImageTags); ok || v != "" {
+		oldList, newList := d.GetChange(isImageTags)
+		err = UpdateTagsUsingCRN(oldList, newList, meta, *image.CRN)
+		if err != nil {
+			log.Printf(
+				"Error on create of resource vpc image (%s) tags: %s", d.Id(), err)
```

Comment:
remove the log statement if used for debug purpose or else use `fmt` for stdout.